build-llvm.py

```diff
@@ -0,0 +1,402 @@
+    # We need it for the LLVMgold plugin, which can be used for LTO with ld.gold,
+    # which at the time of writing this, is how the Google Pixel 3 kernel is built
+    # and linked.
+    utils.download_binutils(root)
+
+
+# Clean up and create the build folder
+def cleanup(build, incremental):
+    if not incremental and build.is_dir():
+        shutil.rmtree(build.as_posix())
+    build.mkdir(parents=True, exist_ok=True)
+
+
+# Invoke cmake to generate the build files
+def invoke_cmake(build, cc, cxx, debug, install_folder, ld, projects, root,
+                 targets):
+    utils.header("Configuring LLVM")
+
+    # Base cmake defintions, which don't depend on any user supplied options
+    defines = {}
+    # Objective-C Automatic Reference Counting (we don't use Objective-C)
+    defines['CLANG_ENABLE_ARCMT'] = 'OFF'
+    # We don't (currently) use the static analyzer
+    defines['CLANG_ENABLE_STATIC_ANALYZER'] = 'OFF'
+    # We don't use the plugin system and this saves cycles according to Chromium OS
+    defines['CLANG_PLUGIN_SUPPORT'] = 'OFF'
+    # The C compiler to use
+    defines['CMAKE_C_COMPILER'] = cc
+    # The C++ compiler to use
+    defines['CMAKE_CXX_COMPILER'] = cxx
+    # Where the toolchain should be installed
+    defines['CMAKE_INSTALL_PREFIX'] = install_folder.as_posix()
+    # For LLVMgold.so, which is used for LTO with ld.gold
+    defines['LLVM_BINUTILS_INCDIR'] = root.joinpath(utils.current_binutils(),
+                                                    "include").as_posix()
+    # The projects to build
+    defines['LLVM_ENABLE_PROJECTS'] = projects
+    # Don't build bindings; they are for other languages that the kernel does not use
+    defines['LLVM_ENABLE_BINDINGS'] = 'OFF'
+    # Don't build Ocaml documentation
+    defines['LLVM_ENABLE_OCAMLDOC'] = 'OFF'
+    # Removes system dependency on terminfo and almost every major clang provider turns this off
+    defines['LLVM_ENABLE_TERMINFO'] = 'OFF'
+    # Don't build clang-tools-extras to cut down on build targets (about 400 files or so)
+    defines['LLVM_EXTERNAL_CLANG_TOOLS_EXTRA_SOURCE_DIR'] = ''
+    # Don't include documentation build targets because it is available on the web
+    defines['LLVM_INCLUDE_DOCS'] = ''
+    # Don't include example build targets to save on cmake cycles
+    defines['LLVM_INCLUDE_EXAMPLES'] = 'OFF'
+    # The architectures to build backends for
+    defines['LLVM_TARGETS_TO_BUILD'] = targets
+
+    # If a debug build was requested
+    if debug:
+        defines['CMAKE_BUILD_TYPE'] = 'Debug'
+        defines['CMAKE_C_FLAGS'] = '-march=native -mtune=native'
+        defines['CMAKE_CXX_FLAGS'] = '-march=native -mtune=native'
+        defines['LLVM_BUILD_TESTS'] = 'ON'
+    # If a release build was requested
+    else:
+        defines['CMAKE_BUILD_TYPE'] = 'Release'
+        defines['CMAKE_C_FLAGS'] = '-O2 -march=native -mtune=native'
+        defines['CMAKE_CXX_FLAGS'] = '-O2 -march=native -mtune=native'
```

Comment:
just curious if `-O3` doesn't work?

Reply:
It does. I don't know if I am just just paranoid but I feel like the toolchain is slower when compiled with `-O3`